shared/podman/utils.go

```diff
@@ -155,21 +158,62 @@ func DeleteVolume(name string, dryRun bool) error {
 			log.Info().Msgf(L("Would run %s"), "podman volume rm "+name)
 		} else {
 			log.Info().Msgf(L("Run %s"), "podman volume rm "+name)
-			err := utils.RunCmd("podman", "volume", "rm", name)
-			if err != nil {
-				log.Error().Err(err).Msgf(L("Failed to remove volume %s"), name)
+			if err := utils.RunCmd("podman", "volume", "rm", name); err != nil {
+				log.Trace().Err(err).Msgf("podman volume rm %s", name)
+				// Check if the volume is not mounted - for example var-pgsql - as second storage device
+				// We need to compute volume path ourselves because above `podman volume rm` call may have
+				// already removed volume from podman internal structures
+				basePath, errBasePath := getPodmanVolumeBasePath()
+				if errBasePath != nil {
+					return errBasePath
+				}
+				target := path.Join(basePath, name)
+				if isVolumePathEmpty(target) && isVolumePathMounted(target) {
+					log.Info().Msgf(L("Volume %s is externally mounted, directory cannot be removed"), name)
+					return nil
+				}
+				return err
 			}
 		}
 	}
 	return nil
 }
 
 func isVolumePresent(volume string) bool {
+	var exitError *exec.ExitError
 	cmd := exec.Command("podman", "volume", "exists", volume)
-	if err := cmd.Run(); err != nil {
+	if err := cmd.Run(); err != nil && errors.As(err, &exitError) {
+		log.Debug().Err(err).Msgf("podman volume exists %s", volume)
 		return false
 	}
-	return cmd.ProcessState.ExitCode() == 0
+	return cmd.ProcessState.Success()
+}
+
+func isVolumePathMounted(volume string) bool {
+	cmd := exec.Command("findmnt", "--target", volume)
+	var exitError *exec.ExitError
+	if err := cmd.Run(); err != nil && errors.As(err, &exitError) {
+		log.Debug().Err(err).Msgf("findmnt --target %s", volume)
+		return false
+	}
+	return cmd.ProcessState.Success()
+}
+
+func isVolumePathEmpty(volume string) bool {
+	f, err := os.Open(volume)
+	if err != nil {
+		return false
+	}
+	defer f.Close()
+
+	_, err = f.Readdirnames(1)
+	return err == io.EOF
+}
+
+func getPodmanVolumeBasePath() (string, error) {
+	cmd := exec.Command("podman", "system", "info", "--format={{ .Store.VolumePath }}")
+	out, err := cmd.Output()
+	return strings.TrimSuffix(string(out), "\n"), err
```

Comment:
I think `TrimSpaces` is good enough for this:

```suggestion
	return strings.TrimSpaces(string(out)), err
```